src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+                //     public override string Property { set { } } // 3
+                Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("DerivedDerivedWithoutInit.Property", "DerivedGetterOnly.Property").WithLocation(35, 28)
+                );
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void OverridingInitOnlyProperty_Metadata(bool emitAsImage)
+        {
+            string lib_cs = @"
+public class Base
+{
+    public virtual string Property { get; init; }
+}";
+            var libComp = CreateCompilation(new[] { lib_cs, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            libComp.VerifyDiagnostics();
+
+            string source = @"
+public class DerivedWithInit : Base
+{
+    public override string Property { get; init; }
+}
+public class DerivedWithoutInit : Base
+{
+    public override string Property { get; set; } // 1
+}
+public class DerivedWithInitSetterOnly : Base
+{
+    public override string Property { init { } }
+}
+public class DerivedWithoutInitSetterOnly : Base
+{
+    public override string Property { set { } } // 2
+}
+public class DerivedGetterOnly : Base
+{
+    public override string Property { get => null; }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview,
+                references: new[] { emitAsImage ? libComp.EmitToImageReference() : libComp.ToMetadataReference() });
+            comp.VerifyDiagnostics(
+                // (8,28): error CS8803: 'DerivedWithoutInit.Property' must match by init-only of overridden member 'Base.Property'
+                //     public override string Property { get; set; } // 1
+                Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("DerivedWithoutInit.Property", "Base.Property").WithLocation(8, 28),
+                // (16,28): error CS8803: 'DerivedWithoutInitSetterOnly.Property' must match by init-only of overridden member 'Base.Property'
+                //     public override string Property { set { } } // 2
+                Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("DerivedWithoutInitSetterOnly.Property", "Base.Property").WithLocation(16, 28)
+                );
+        }
+
+        [Fact]
+        public void OverridingRegularProperty()
+        {
+            string source = @"
+public class Base
+{
+    public virtual string Property { get; set; }
+}
+public class DerivedWithInit : Base
+{
+    public override string Property { get; init; } // 1
+}
+public class DerivedWithoutInit : Base
+{
+    public override string Property { get; set; }
+}
+public class DerivedWithInitSetterOnly : Base
+{
+    public override string Property { init { } } // 2
+}
+public class DerivedWithoutInitSetterOnly : Base
+{
+    public override string Property { set { } }
+}
+public class DerivedGetterOnly : Base
```

Comment:
>DerivedGetterOnly [](start = 13, length = 17)

Is scenario similar to the third scenario from OverridingInitOnlyProperty not interesting?